benchmarks/ASQA/README.md

```diff
@@ -1,55 +1,141 @@
-2. Run `run.sh` to start dataset generation. The result will be saved in `output` directory, named `results.json`. And the `result_datasets.json` file is the json dump of the detailed result.
+Replace api_key to your OpenAI api key in `run_generate.sh` then run it to generate `gpt-3.5-turbo-instruct` response. The command is as follows:
 
-### Arguments:
+```bash
+python3 benchmarks/ASQA/generate.py \
+        --max_num_examples 500 \
+        --max_new_tokens 256 \
+        --output_path "benchmarks/ASQA/output" \
+        --model "gpt-3.5-turbo-instruct" \
+        --api_key "YOUR_API_KEY" 
+```
+
+Arguements:
+
+- `--max_num_examples`: The maximum number of examples used in generate answers.
+- `--max_new_tokens`: The maximum number of tokens that can be generated for answering questions.
+- `--output_path`: Directory that the generated answers will be saved.
+- `--model`: The OpenAI GPT model to use, e.g., gpt-3.5-turbo-instruct. And result file will be named as `f"{model}.jsonl"`.
+- `--api_key`: Your OpenAI API key.
+
+### 4.2 Evaluation
 
-`--max_num_examples`: The maximum number of examples used in generate answers.
+1. Prepare RAG responses. By default, `asqa_benchmark.py` will download the results of the `gpt-3.5-turbo-instruct` model from [our huggingface dataset](https://huggingface.co/datasets/golaxy/rag-bench) and evaluate them. If you wish to evaluate your own results, you can simply attach your predictions as `answers` to the end of each example in the original ASQA dataset, similar to what we did in [the file](data\gpt-3.5-turbo-instruct.jsonl).
 
-`--max_new_tokens`: The maximum number of tokens that can be generated for answering questions.
+2. Evaluate the responses. Run `run.sh` to start dataset evaluation. The results will be saved in the output directory, named `results.jsonl`. Additionally, the `result_datasets.jsonl` file is a JSON dump of the detailed results, including scores for every example in the dataset.The command is as follows:
 
-`--output_dir`: Directory where the generated answers and the final results will be saved.
+```bash
+python3 benchmarks/ASQA/asqa_benchmark.py \
+    --output_dir "benchmarks/ASQA/output" \
+    --dataset_path "benchmarks/ASQA/data/gpt-3.5-turbo-instruct.jsonl"
+```
+
+Arguements:
 
-`--model`: The OpenAI GPT model to use, e.g., gpt-3.5-turbo-instruct.
+- `--output_dir`: Output directory to save results.
+- `dataset_path`: The dataset including model predictions and ground truths to evaluate.
 
-`--api_key`: Your OpenAI API key.
 
-## Metrics
+## 5. Performance
 
-## Performance
+Here are results of different models.
 
-| Model | Rouge-L | Exact Match | Disambig F1 | D-R Score|
+| Model | STR-EM | Rouge-L | Disambig F1 | D-R Score|
 |:---:|:---:|:---:|:---:|:---:|
-| gpt-3.5-turbo-instruct | 29.8 | 34.0 | 30.6 | 30.2 |
-| text-davinci-003 (FLARE) | 33.3 | 33.8 | 24.2 | 28.4 |
+| gpt-3.5-turbo-instruct | 33.8 | 30.2 | 30.7 | 30.5 |
```

Comment:
add huggingface link~

Reply:
Instance level result could be found in [link], [link], and [link], respectively.